modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -0,0 +1,427 @@
+                                      all_ndims, orig_shapes, orig_steps,
+                                      shapes, steps))
+            return;
+
+        do_broadcast_op<T, Functor>(
+                    max_ndims, shapes[0], a.ptr<char>(), steps[1],
+                    b.ptr<char>(), steps[2], out.ptr<char>(), steps[0],
+                    f);
+    }
+
+    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        if (inputs_arr.depth() == CV_16S)
+        {
+            forward_fallback(inputs_arr, outputs_arr, internals_arr);
+            return;
+        }
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        Mat& a = inputs[0];
+        Mat& b = inputs[1];
+        Mat& out = outputs[0];
+        CV_Assert(a.type() == b.type() && b.type() == out.type());
+
+        typeDispatch(a.type(), a, b, out);
+    }
+
+    template<typename T, typename... Args>
+    inline void opDispatch(Args&&... args)
+    {
+        switch (op)
+        {
+            case OPERATION::EQUAL:
+            {
+                auto equal = [](const T &a, const T &b) { return a == b; };
+                forward_impl<T>(equal, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::GREATER:
+            {
+                auto greater = [](const T &a, const T &b) { return a > b; };
+                forward_impl<T>(greater, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::GREATER_EQUAL:
+            {
+                auto greater_equal = [](const T &a, const T &b) { return a >= b; };
+                forward_impl<T>(greater_equal, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::LESS:
+            {
+                auto less = [](const T &a, const T &b) { return a < b; };
+                forward_impl<T>(less, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::LESS_EQUAL:
+            {
+                auto less_equal = [](const T &a, const T &b) { return a <= b; };
+                forward_impl<T>(less_equal, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::POW:
+            {
+                auto pow = [] (const T& a, const T& b) { return std::pow(a, b); };
+                forward_impl<T>(pow, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::BITSHIFT:
+            {
+                auto bitshift = [] (const uint8_t &a, const uint8_t &b) { return a << b; };
+                forward_impl<T>(bitshift, std::forward<Args>(args)...);
+                break;
+            }
+            case OPERATION::MAX:
```

Comment:
by ONNX specification, Max, Min, Mean and Sum may have more than 2 inputs. Is it planned to be added in this PR or later?

Reply:
Yes, it is planned, this is the reason this PR is still in draft.